src/ralph/api/routers/statements.py

```diff
@@ -423,26 +423,64 @@ async def put(
     _enrich_statement_with_authority(statement_as_dict, current_user)
 
     try:
-        existing_statements = desync(
-            BACKEND_CLIENT.query_statements_by_ids([statementId])
-        )
+        existing_statements = BACKEND_CLIENT.query_statements_by_ids([statementId])
     except BackendException as error:
         raise HTTPException(
             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
             detail="xAPI statements query failed",
         ) from error
 
-    if existing_statements:
-        # The LRS specification calls for deep comparison of duplicate statement ids.
-        # In the case that the current statement is not equivalent to one found
-        # in the database we return a 409, otherwise the usual 204.
-        async for existing in existing_statements:
-            if not statements_are_equivalent(statement_as_dict, existing):
-                raise HTTPException(
-                    status_code=status.HTTP_409_CONFLICT,
-                    detail="A different statement already exists with the same ID",
-                )
-        return
+
+    if isasyncgen(existing_statements):
+        first_statement = None
+        try:
+            # Check if existing_statements is not empty
+            first_statement = await anext(existing_statements)
+        except StopIteration as exc:
+            logger.debug("No statements with given ID exist: %s", exc)
+            pass
+
+        if first_statement is not None:
+            existing_statements = chain((first_statement,), existing_statements)
+            # The LRS specification calls for deep comparison of duplicate statement ids.
+            # In the case that the current statement is not equivalent to one found
+            # in the database we return a 409, otherwise the usual 204.
+            async for existing in existing_statements:
+                if not statements_are_equivalent(statement_as_dict, existing):
+                    raise HTTPException(
+                        status_code=status.HTTP_409_CONFLICT,
+                        detail="A different statement already exists with the same ID",
+                    )
+            return
+        else:
+            pass
```

Comment:
```suggestion
        try:
            # Check if existing_statements is not empty
            first_statement = await anext(existing_statements)
        except StopIteration as exc:
            logger.debug("No statements with given ID exist: %s", exc)
            first_statement = None
            pass

        if not first_statement:
            pass
            
        existing_statements = chain((first_statement,), existing_statements)
        # The LRS specification calls for deep comparison of duplicate statement ids.
        # In the case that the current statement is not equivalent to one found
        # in the database we return a 409, otherwise the usual 204.
        async for existing in existing_statements:
            if not statements_are_equivalent(statement_as_dict, existing):
                raise HTTPException(
                    status_code=status.HTTP_409_CONFLICT,
                    detail="A different statement already exists with the same ID",
                )
        return
```